test/cases/edge.sh

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/bash
+
+#
+# Edge Rest API integration test
+#
+# The test flow in this file is:
+# ------------ Test commit image --------------------------------------------------------
+# 1. Call image builder Rest API. (build an rhel-edge-commit image and upload to aws s3)
+# 2. Download commit image, extract it to /var/www/html, and serve it over httpd
+# 3. Install Edge vm based on commit repo url, and run ansible playbook to check it.
+#
+# ------------ Test installer image ------------------------------------------------------
+# 4. Call image builder Rest API. (build an rhel-edge-install image and upload to aws s3)
+# 4. Download installer image.
+# 5. Install Edge vm with the installer image,and run ansible playbook to check it.
+
+set -euxo pipefail
+
+############### Common variables for CI ###########################
+WORKDIR=$(mktemp -d)
+IMAGE_BUILDER_TEST_DATA=/usr/share/tests/image-builder
+
+############### Common variables for image builder ################
+PORT="8086"
+CURLCMD='curl -w %{http_code}'
+HEADER="x-rh-identity: eyJlbnRpdGxlbWVudHMiOnsiaW5zaWdodHMiOnsiaXNfZW50aXRsZWQiOnRydWV9LCJzbWFydF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwib3BlbnNoaWZ0Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwiaHlicmlkIjp7ImlzX2VudGl0bGVkIjp0cnVlfSwibWlncmF0aW9ucyI6eyJpc19lbnRpdGxlZCI6dHJ1ZX0sImFuc2libGUiOnsiaXNfZW50aXRsZWQiOnRydWV9fSwiaWRlbnRpdHkiOnsiYWNjb3VudF9udW1iZXIiOiIwMDAwMDAiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJ1c2VyIiwiZW1haWwiOiJ1c2VyQHVzZXIudXNlciIsImZpcnN0X25hbWUiOiJ1c2VyIiwibGFzdF9uYW1lIjoidXNlciIsImlzX2FjdGl2ZSI6dHJ1ZSwiaXNfb3JnX2FkbWluIjp0cnVlLCJpc19pbnRlcm5hbCI6dHJ1ZSwibG9jYWxlIjoiZW4tVVMifSwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMDAwMDAwIn19fQ=="
+ADDRESS="localhost"
+BASEURL="http://$ADDRESS:$PORT/api/image-builder/v1.0"
+ARCH=$(uname -m)
+
+############### Common variables for Edge ################
+HTTPD_PATH="/var/www/html"
+TEST_UUID=$(uuidgen)
+IMAGE_KEY="edge-${TEST_UUID}"
+IMAGE_TYPE=rhel-edge
+OS_VARIANT="rhel8-unknown"
+OSTREE_REF="rhel/8/${ARCH}/edge"
+BOOT_LOCATION="http://download.devel.redhat.com/rel-eng/rhel-8/RHEL-8/latest-RHEL-8.4.0/compose/BaseOS/x86_64/os/"
+HOST_ADDRESS=192.168.100.1
+GUEST_ADDRESS=192.168.100.50
+REPO_URL=http://$HOST_ADDRESS/repo
+
+SSH_OPTIONS=(-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5)
+SSH_KEY=${IMAGE_BUILDER_TEST_DATA}/keyring/id_rsa
```

Comment:
Instead of hard-coding the key in git can we instead generate it during test setup ?

Reply:
Good suggestion, thank you!
Can we implement it next time? Because we have been using same ssh key for both upstream and downstream testing for a long time, makes it easy to debug the edge issues. Other people in our team uses the same key.

Reply:
https://issues.redhat.com/browse/COMPOSER-1011
https://issues.redhat.com/browse/COMPOSER-1012